src/modules/landing_target_estimator/LandingTargetEstimator.cpp

```diff
@@ -50,260 +51,1431 @@
+			target_gps_lon = target_GNSS_report.lon;
+			target_gps_alt = target_GNSS_report.alt;
+
+			// TODO: complete mavlink message to include uncertainties.
+			// gps_target_eph = target_GNSS_report.cov_x_rel;
+			// gps_target_epv = target_GNSS_report.cov_z_rel;
+
+			gps_target_eph = _gps_target_unc;
+			gps_target_epv = _gps_target_unc;
 		}
 
-		if (!PX4_ISFINITE(_irlockReport.pos_y) || !PX4_ISFINITE(_irlockReport.pos_x)) {
-			return;
+	} else if ((_ltest_aid_mask & SensorFusionMask::USE_MISSION_POS)
+		   && _nave_state == vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION && _landing_pos.valid) {
+
+		use_gps_measurements = true;
+
+		// Measurement of the landing target position comes from the mission item only
+		target_gps_lat = _landing_pos.lat;
+		target_gps_lon = _landing_pos.lon;
+		target_gps_alt = _landing_pos.alt;
+
+		gps_target_eph = _gps_target_unc;
+		gps_target_epv = _gps_target_unc;
+	}
+
+	if (use_gps_measurements) {
+
+		// Obtain GPS relative measurements in NED as target_global - uav_gps_global followed by global2local transformation
+		Vector3f gps_relative_pos;
+		get_vector_to_next_waypoint((vehicle_gps_position.lat / 1.0e7), (vehicle_gps_position.lon / 1.0e7),
+					    (target_gps_lat / 1.0e7), (target_gps_lon / 1.0e7),
+					    &gps_relative_pos(0), &gps_relative_pos(1));
+
+		// Down direction (if the drone is above the target, the relative position is positive)
+		gps_relative_pos(2) = (vehicle_gps_position.alt - target_gps_alt) / 1000.f; // transform mm to m
+
+		// Var(aX - bY) = a^2 Var(X) + b^2Var(Y) - 2ab Cov(X,Y)
+		float gps_unc_horizontal = vehicle_gps_position.eph + gps_target_eph;
+		float gps_unc_vertical = vehicle_gps_position.epv + gps_target_epv;
+
+		// GPS already in NED, no rotation required. STATE: [pose,vel,bias,acc]
+		if (_target_model == TargetModel::FullPoseCoupled) {
+			// State: [rx, ry, rz, r_x_dot, r_y_dot, r_z_dot, bx, by, bz, ... ]
+
+			// x direction H = [1, 0, 0, 0, 0, 0, 1, 0, 0, ...]
+			obs->meas_h_xyz(0, 0) = 1;
+			obs->meas_h_xyz(0, 6) = 1;
+
+			// y direction H = [0, 1, 0, 0, 0, 0, 0, 1, 0, ...]
+			obs->meas_h_xyz(1, 1) = 1;
+			obs->meas_h_xyz(1, 7) = 1;
+
+			// z direction H = [0, 0, 1, 0, 0, 0, 0, 0, 1, ...]
+			obs->meas_h_xyz(2, 2) = 1;
+			obs->meas_h_xyz(2, 8) = 1;
+
+		} else {
+			// State: [r, r_dot, b, ...] --> same for x,y,z directions (decoupled)
+			obs->meas_h_xyz(0, 0) = 1;
+			obs->meas_h_xyz(0, 2) = 1;
+
+			obs->meas_h_xyz(1, 0) = 1;
+			obs->meas_h_xyz(1, 2) = 1;
+
+			obs->meas_h_xyz(2, 0) = 1;
+			obs->meas_h_xyz(2, 2) = 1;
+		}
+
+		obs->timestamp = gps_timestamp;
+
+		obs->meas_xyz = gps_relative_pos;
+
+		obs->meas_unc_xyz(0) = gps_unc_horizontal;
+		obs->meas_unc_xyz(1) = gps_unc_horizontal;
+		obs->meas_unc_xyz(2) = gps_unc_vertical;
+
+		obs->updated_xyz.setAll(true);
+
+		return true;
+	}
+
+	return false;
+
+}
+
+bool LandingTargetEstimator::processObsUWB(const uwb_distance_s uwb_distance, targetObsPos *obs)
```

Comment:
```suggestion
bool LandingTargetEstimator::processObsUWB(const uwb_distance_s uwb_distance, targetObsPos &obs)
```
use references instead of pointers if possible

Reply:
Done in https://github.com/JonasPerolini/PX4-Autopilot/commit/ea0c0113b1a7fbe7f4465e657dd69ad93a59c675